File: WorldFeed/Server/src/WorldFeed.Common/Public/Exceptions/TwitterInvalidCredentialsException.cs
namespace WorldFeed.Common.Public.Exceptions
{
    using System;

    using WorldFeed.Common.Models.Authentication;

    /// <summary>
    /// Exception raised when attempting to perform operations without the proper credentials
    /// </summary>
    public class TwitterInvalidCredentialsException : Exception
    {

        public IReadOnlyConsumerCredentials Credentials { get; }

        public TwitterInvalidCredentialsException(string message): base(message)
        {
        }

        public TwitterInvalidCredentialsException(IReadOnlyConsumerCredentials credentials)
            : base("The consumer key and consumer secret must be defined!")
        {
            Credentials = credentials;
        }
    }
}

File: Sharebook/Client/src/app/core/Public/Exceptions/TwitterInvalidCredentialsException.ts
import {IReadOnlyConsumerCredentials} from "../../Core/Models/Authentication/ReadOnlyConsumerCredentials";
import Exception from "typescript-dotnet-commonjs/System/Exception";

// Exception raised when attempting to perform operations without the proper credentials
export class TwitterInvalidCredentialsException extends Exception {
  constructor(message: string, credentials?: IReadOnlyConsumerCredentials) {
    if (credentials) {
      super("The consumer key and consumer secret must be defined!");

      this.credentials = credentials;
    } else {
      super(message);
    }
  }

  public credentials: IReadOnlyConsumerCredentials;
}

// public TwitterInvalidCredentialsException(message: string): base(message)
// {
// }
//
// public TwitterInvalidCredentialsException(IReadOnlyConsumerCredentials credentials)
// : base("The consumer key and consumer secret must be defined!")
// {
//   Credentials = credentials;
// }
